contracts/src/AgentExecutor.sol

```diff
@@ -16,11 +17,15 @@ contract AgentExecutor {
 
     /// @dev Transfer ether to `recipient`. Unlike `_transferToken` This logic is not nested within `execute`,
     /// as the gateway needs to control an agent's ether balance directly.
-    ///
     function transferNative(address payable recipient, uint256 amount) external {
         recipient.safeNativeTransfer(amount);
     }
 
+    /// @dev Transfer ether to Gateway. Used once off for migration purposes. Can be removed after version 1.
+    function transferNativeToGateway(address payable gateway, uint256 amount) external {
```

Comment:
In this context, `msg.sender` will always be the gateway address, as verified by `Agent.invoke`. So having a `gateway` parameter is redundant.

Also, prefer this naming in this context:
```suggestion
    function transferEtherToGateway(address payable gateway, uint256 amount) external {
```

Reply:
Addressed in aca38466c8d25da51ecf7629014f20b9f47d8871